File: Submission/Week 11/Clock.py
class Counter:
    def __init__(self, name):
        self.__name = name
        self.__count = 0
    def Incremend(self):
        self.__count += 1
    def Reset(self):
        self.__count = 0

    @property
    def Ticks(self):
        return self.__count

    @property 
    def Name(self):
        return self.__name
    @Name.setter
    def Name(self, new_name):
        self.__name = new_name

class Clock:
    def __init__(self):
        self.__second = Counter("Second")
        self.__minute = Counter("Minute")
        self.__hour = Counter("Hour")

    def Tick(self):
        self.__second.Incremend()
        if self.__second.Ticks == 60:
            self.__minute.Incremend()
            self.__second.Reset()
            if self.__minute.Ticks == 60:
                self.__hour.Incremend()
                self.__minute.Reset()

    @property 
    def ReadTime(self):
        return f'{self.__hour.Ticks:02d}:{self.__minute.Ticks:02d}:{self.__second.Ticks:02d}'

    def ResetClock(self):
        self.__second.Reset()
        self.__minute.Reset()
        self.__hour.Reset()


def mult(n,x):
    if x == 1:
        return n
    else:
        return n + mult(n, x-1)

print(mult(5, 5))

File: Submission/Week 3/Clock/CounterClock/Clock.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CounterClock
{
    public class Clock
    {
        private Counter _second;
        private Counter _minute;
        private Counter _hour;
        public Clock()
        {

            _second = new Counter("seconds");
            _minute = new Counter("minutes");
            _hour = new Counter("hours");

        }

        public void Tick()
        {
            _second.Incremend();
            if (_second.Ticks == 60)
            {
                _minute.Incremend();
                _second.Reset();
                if (_minute.Ticks == 60)
                {
                    _hour.Incremend();
                    _minute.Reset();
                }
            }
        }

        public string ReadTime
        {
            get
            {
                return $"{_hour.Ticks.ToString("00")}:{_minute.Ticks.ToString("00")}:{_second.Ticks.ToString("00")}";
            }
        }

        public void ResetClock()
        {
            _second.Reset();
            _minute.Reset();
            _hour.Reset();
        }
    }
}
